avocado/utils/softwareraid.py

```diff
@@ -0,0 +1,161 @@
+    :type metadata: str
+    :param spare: List of spare disks for software raid
+    :type spare: list
+    """
+    def __init__(self, name, level, disk, metadata, spare=''):
+        self.name = name
+        self.level = level
+        self.metadata = metadata
+        self.disk = disk
+        self.remadd = ''
+        self.spare = spare
+
+    def create(self):
+        """
+        Creates software raid.
+
+        :return: True if raid is created. False otherwise.
+        :rtype: bool
+        """
+        cmd = "mdadm --create --assume-clean %s" % self.name
+        cmd += " --level=%s" % self.level
+        cmd += " --raid-devices=%d %s" % (len(self.disk),
+                                          " ".join(self.disk))
+        cmd += " --metadata=%s" % self.metadata
+        if self.spare:
+            cmd += " --spare-devices=%d %s" % (len(self.spare),
+                                               " ".join(self.spare))
+        cmd += " --verbose --force"
+        return self._run_command(cmd)
+
+    def _run_command(self, cmd, log_details=True):
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            if log_details:
+                log.debug(self.get_detail())
+            return False
+        if log_details:
+            log.debug(self.get_detail())
+        return True
+
+    def get_detail(self):
+        """
+        Returns mdadm details.
+
+        :return: mdadm --detail output
+        :rtype: str
+        """
+        cmd = "mdadm --detail %s" % self.name
+        output = process.run(cmd, ignore_status=True, shell=True)
+        return output.stdout_text
+
+    def is_recovering(self):
+        """
+        Checks if raid is recovering.
+
+        :return: True if recovering, False otherwise.
+        :rtype: bool
+        """
+        for line in self.get_detail().splitlines():
+            if 'State' in line and 'recovering' in line:
+                return True
+        return False
+
+    def remove(self, disk):
+        """
+        Removes disk specified from software raid.
+
+        :param disk: disk to be removed.
+        :type disk: str
+
+        :return: True if remove is successful, False otherwise.
+        rtype: bool
+        """
+        cmd = "mdadm %s --fail %s --remove %s" % (self.name, disk, disk)
+        return self._run_command(cmd)
+
+    def add(self, disk):
+        """
+        Adds disk specified to software raid.
+
+        :param disk: disk to be added.
+        :type disk: str
+
+        :return: True if add is successful, False otherwise.
+        rtype: bool
+        """
+        cmd = "mdadm %s --add %s" % (self.name, disk)
+        return self._run_command(cmd)
+
+    def stop(self):
+        """
+        Stops software raid.
+
+        :return: True if stopped, False otherwise.
+        :rtype: bool
+        """
+        cmd = "mdadm --manage %s --stop" % self.name
+        return self._run_command(cmd)
+
+    def assemble(self):
```

Comment:
Also, I would like to suggest to reorder the methods in alphabetic order, it is easy to maintain the code.